src/dns.rs

```diff
@@ -24,18 +24,23 @@ impl Resolver {
     }
 
     pub async fn lookup_ip(self, name: String) -> Result<LookupIp, DnsError> {
-        spawn_blocking(move || {
-            // We need to add port with the name so that `to_socket_addrs`
-            // resolves it properly. We will be discarding the port afterwards.
-            //
-            // Any port will do, but `9` is a well defined port for discarding
-            // packets.
-            (name.as_ref(), 9).to_socket_addrs()
-        })
-        .await
-        .context(JoinError)?
-        .map(LookupIp)
-        .context(UnableLookup)
+        // We need to add port with the name so that `to_socket_addrs`
+        // resolves it properly. We will be discarding the port afterwards.
+        //
+        // Any port will do, but `9` is a well defined port for discarding
+        // packets.
+        let dummy_port = 9;
+        if name == "localhost" {
```

Comment:
This is great. Would be good to a ref to https://tools.ietf.org/html/rfc6761#section-6.3 for clarity.

Reply:
Done.